include/bluetooth/bluetooth.h

```diff
@@ -775,6 +817,236 @@ int bt_le_ext_adv_get_info(const struct bt_le_ext_adv *adv,
+ * advertising.
+ *
+ * @param adv      Advertising set object.
+ *
+ * @return         Zero on success or (negative) error code otherwise.
+ */
+int bt_le_per_adv_stop(struct bt_le_ext_adv *adv);
+
+struct bt_le_per_adv_sync_synced_info {
+	/** Advertiser LE address and type. */
+	const bt_addr_le_t *addr;
+
+	/** Periodic advertising interval (N * 1.25MS) */
+	uint16_t interval;
+
+	/** Advertiser PHY */
+	uint8_t phy;
+};
+
+struct bt_le_per_adv_sync_term_info {
+	/** Advertiser LE address and type. */
+	const bt_addr_le_t *addr;
+};
+
+
+/** Constant Tone Extension (CTE) types */
+enum {
+	/** Angle of Arrival */
+	BT_LE_PER_ADV_CTE_AOA = 0x00,
+	/** Angle of Departure with 1 us slots */
+	BT_LE_PER_ADV_CTE_AOD_1US = 0x01,
+	/** Angle of Departure with 2 us slots */
+	BT_LE_PER_ADV_CTE_AOD_2US = 0x02,
+	/** No extensions */
+	BT_LE_PER_ADV_CTE_NONE = 0xFF,
+};
+
+struct bt_le_per_adv_sync_recv_info {
+	/** Advertiser LE address and type. */
+	const bt_addr_le_t *addr;
+
+	/** The TX power of the advertisement. */
+	int8_t tx_power;
+
+	/** The RSSI of the advertisement excluding any CTE. */
+	int8_t rssi;
+
+	/** The Constant Tone Extension (CTE) of the advertisement */
+	uint8_t cte_type;
+};
+
+struct bt_le_per_adv_sync_cb {
+	/** @brief The periodic advertising has been successfully synced.
+	 *
+	 *  This callback notifies the application that the periodic advertising
+	 *  set has been successfully synced, and will now start to
+	 *  receive periodic advertising reports.
+	 *
+	 *  @param sync The periodic advertising sync object.
+	 *  @param info Information about the sync event.
+	 */
+	void (*synced)(struct bt_le_per_adv_sync *sync,
+		       struct bt_le_per_adv_sync_synced_info *info);
+
+	/** @brief The periodic advertising sync has been terminated.
+	 *
+	 *  This callback notifies the application that the periodic advertising
+	 *  sync has been terminated, either by local request, remote request or
+	 *  because due to missing data (e.g. by being out of range or sync).
+	 *
+	 *  @param sync  The periodic advertising sync object.
+	 */
+	void (*term)(struct bt_le_per_adv_sync *sync,
+			   const struct bt_le_per_adv_sync_term_info *info);
+
+	/** @brief Periodic advertising data received.
+	 *
+	 *  This callback notifies the application of an periodic advertising
+	 *  report.
+	 *
+	 *  @param sync  The advertising set object.
+	 *  @param info  Information about the periodic advertising event.
+	 *  @param buf   Buffer containing the periodic advertising data.
+	 */
+	void (*recv)(struct bt_le_per_adv_sync *sync,
+		     const struct bt_le_per_adv_sync_recv_info *info,
+		     struct net_buf_simple *buf);
+};
+
+/** Periodic advertising sync options */
+enum {
+	/** Convenience value when no options are specified. */
+	BT_LE_PER_ADV_SYNC_OPT_NONE = 0,
+
+	/** Use the periodic advertising list to sync with advertiser */
+	BT_LE_PER_ADV_SYNC_OPT_USE_PER_ADV_LIST = BIT(0),
```

Comment:
No API for adding devices to the advertising list. Will you add it later?

Reply:
That was the plan, yes. Figured this PR would be way too large if it contained all the PA features. Another feature to come are PA sync transfer (PAST).